docs-src/references/web-ui-environment-variables.md

```diff
@@ -0,0 +1,227 @@
+---
+id: web-ui-environment-variables
+title: Temporal Web UI environmental variables
+sidebar_label: Web UI env var
+description: How to set  environmental variables for Temporal Web UI.
+tags:
+  - docker
+  - webui
+  - ui server
+  - reference
+---
+
+You can use environment variables to dynamically alter the configuration of your Temporal Web UI.
+
+These can be used in many environments such as Docker.
+For example:
+
+<!-- ```
+docker run \
+    -e TEMPORAL_ADDRESS=127.0.0.1:7233 \
+    -e TEMPORAL_UI_PORT=8080 \
+    -e TEMPORAL_AUTH_ENABLED=true \
+    -e TEMPORAL_AUTH_PROVIDER_URL=https://accounts.google.com \
+    -e TEMPORAL_AUTH_CLIENT_ID=xxxxx-xxxx.apps.googleusercontent.com \
+    -e TEMPORAL_AUTH_CLIENT_SECRET=xxxxxxxxxxxxxxx \
+    -e TEMPORAL_AUTH_CALLBACK_URL=https://xxxx.com:8080/auth/sso/callback \
+    -e TEMPORAL_UI_ENABLED=true \
+    -e TEMPORAL_OPENAPI_ENABLED=true \
+    -e TEMPORAL_TLS_CA=../ca.cert \
+    -e TEMPORAL_TLS_CERT=../cluster.pem \
+    -e TEMPORAL_TLS_KEY=../cluster.key \
+    -e TEMPORAL_TLS_ENABLE_HOST_VERIFICATION=true \
+    -e TEMPORAL_TLS_SERVER_NAME=tls-server \
+    temporalio/ui:<tag>
+``` -->
+
+The environment variables are defined in the [UI server configuration template file](https://github.com/temporalio/ui-server/blob/main/docker/config_template.yaml) and described in more detail below.
+
+## `TEMPORAL_ADDRESS`
+
+The [Frontend Service](/concepts/what-is-a-frontend-service) address for the Temporal Cluster.
+This environmental variable can be set [in the base configuration file](/references/web-ui-configuration#temporalgrpcaddress) using `temporalGrpcAddress`.
+
+This variable is required for setting other environmental variables.
+`TEMPORAL_ADDRESS` is also needed when setting up [Cross-Origin Resource Sharing](/references/web-ui-configuration#cors).
+
+## `TEMPORAL_UI_PORT`
+
+The port used by the Temporal WebUI Server and the HTTP API.
+
+Defining this variable allows the Cluster and API to communicate effectively.
+
+This variable is needed for `TEMPORAL_OPENAPI_ENABLED` and all auth-related settings to work properly.
+`TEMPORAL_UI_ENABLED` requires a valid port number.
+
+## `TEMPORAL_PUBLIC_PATH`
+
+The location in the Web UI Server where everything is stored.
+
+## `TEMPORAL_AUTH_ENABLED`
+
+Enables or disables Temporal authentication and authorization methods.
+
+When enabled, the Temporal Cluster will use the provider information in the [UI configuration file](/references/web-ui-configuration#auth) to verify the identity of users.
+
+All auth-related environmental variables depend on `TEMPORAL_AUTH_ENABLED`.
+Disabling the variable will retain given values.
+
+All auth-related variables must be defined when this is enabled.
+
+## `TEMPORAL_AUTH_PROVIDER_URL`
+
+The URL for Temporal's authentication and authorization OIDC provider.
```

Comment:
```suggestion
The .well-known IDP discovery URL for authentication and authorization.
```

Reply:
as an example the common terminology  https://help.akana.com/content/current/cm/api_oauth/oauth_discovery/m_oauth_getOpenIdConnectWellknownConfiguration.htm